### DISCUSSION TITLE
Page script still executes even if middleware with redirection is specified

### DISCUSSION THREAD

**Discussion by victor-duarte** [nuxt/nuxt#16968] [Questions]
👋🏻 All,

It seems I am misunderstanding the intention of the middleware and what it means in the docs by

> Nuxt provides a customizable route middleware framework you can use throughout your application, ideal for extracting code that you want to run before navigating to a particular route

For the following scenario:

Middleware:
```
// middleware/lorem.ts

export default defineNuxtRouteMiddleware((to) => {
   if(some condition) return navigateTo('/');  
});
```

Page:
```
// pages/my-page.vue

<script setup lang="ts">
definePageMeta({
  middleware: 'lorem',
  title: 'my page',
});

console.log('log 1')

onBeforeMount(() => {
  console.log('log 3')
})

console.log('log 2')
</script>

<template>
  <p>Hello World</p>
</template>
```

Assuming that `some condition` is true, when I navigate to /my-page I do get redirect to '/' but I see`log 1` and `log 2` are printed in console (server).

- Should these be logged? meaning the code within the script still gets executed?
- If I am doing an API call it gets called but I would expect to not be called since I am "redirecting" in the middleware, right? 
- If the code in script is expected to be executed, what would be the recommended strategy to prevent for example API calls to happen?

I am missing the piece of the puzzle so any pointers (examples, docs, posts) will be appreciated.

Thank you,

**Comment by danielroe** [comment]
I would not expect that. Would you open an issue with a reproduction?

**Reply by victor-duarte** [reply]
Funny enough I am not able to reproduce on a clean install so it could be something in my code base.
Nevertheless will keep looking into it this week and come back with either a reproduction or a conclusion.

Having said that, https://github.com/nuxt/framework/issues/6237 issue could be related.
@larbish posted a reproduction https://github.com/nuxt/framework/issues/6237#issuecomment-1234087137
